packages/kbn-observable/src/factories/race.test.ts

```diff
@@ -0,0 +1,304 @@
+import { Observable } from '../observable';
+import { $race } from './race';
+import { $of } from './of';
+import { collect } from '../lib/collect';
+import { Subject } from '../subjects';
+import { mergeMap } from '../operators/merge_map';
+
+test('should race a single observable', async () => {
+  const res = await collect($race($of(1, 2, 3)));
+  expect(res).toEqual([1, 2, 3, 'C']);
+});
+
+test('completes immediately if given no observable', async () => {
+  const res = await collect($race());
+  expect(res).toEqual(['C']);
+});
+
+test('should race cold and cold', done => {
+  const obs1 = new Observable<string>(observer => {
+    setTimeout(() => {
+      observer.next('foo');
+      observer.complete();
+    }, 30);
+  });
+  const obs2 = new Observable<string>(observer => {
+    setTimeout(() => {
+      observer.next('bar');
+      observer.complete();
+    }, 60);
+  });
+
+  const values: any[] = [];
+
+  $race(obs1, obs2).subscribe({
+    next(v) {
+      values.push(v);
+    },
+    complete() {
+      values.push('C');
+    },
+  });
+
+  setTimeout(() => {
```

Comment:
__question:__ maybe we should just use`fakeTimers` and `advanceTimersByTime` in all of these tests to not be dependent on the slow CI? Theoretically we shouldn't have issues with these simple tests, but real timeouts in tests always a good recipe for flakiness :)